plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerMetadata.java

```diff
@@ -503,6 +527,58 @@ else if (metadata.getStatisticType().equals(ColumnStatisticType.NUMBER_OF_DISTIN
         }).collect(toImmutableList()));
     }
 
+    private Map<String, List<Object>> getColumnValues(SchemaTableName tableName, TableInfo info, Map<String, Long> distinctValues, Map<String, Object> minimums, Map<String, Object> maximums)
+    {
+        long schemaDictionarySize = (long) getSchema(tableName.getSchemaName()).properties().getOrDefault(SchemaInfo.MAX_DICTIONARY_SIZE, maxDictionarySize);
+        long maxDictionarySize = (long) info.properties().getOrDefault(TableInfo.MAX_DICTIONARY_SIZE, schemaDictionarySize);
+        if (maxDictionarySize == 0) {
+            return Map.of();
+        }
+        List<ColumnInfo> columns = info.columns();
+        Map<String, Type> types = columns.stream().collect(toMap(ColumnInfo::name, ColumnInfo::type));
+        List<String> dictionaryColumns = distinctValues.entrySet().stream()
+                .filter(entry -> entry.getValue() <= maxDictionarySize)
+                .map(Map.Entry::getKey)
+                .collect(toImmutableList());
+        ImmutableMap<String, FakerColumnHandle> columnHandles = info.columns().stream()
+                .collect(toImmutableMap(ColumnInfo::name, column -> column.handle().withNullProbability(0)));
+        ImmutableMap.Builder<String, List<Object>> builder = ImmutableMap.builder();
+        try (FakerPageSource pageSource = new FakerPageSource(
+                faker,
+                random,
+                dictionaryColumns.stream()
+                        .map(columnHandles::get)
+                        .filter(Objects::nonNull)
+                        .map(column -> !minimums.containsKey(column.name()) ? column : column.withDomain(Domain.create(ValueSet.ofRanges(Range.range(
+                                column.type(),
+                                minimums.get(column.name()),
+                                true,
+                                maximums.get(column.name()),
+                                true)), false)))
+                        .collect(toImmutableList()),
+                0,
+                maxDictionarySize * 2)) {
+            // TODO support reading values from multiple pages
+            Page page = null;
+            while (page == null) {
+                page = pageSource.getNextPage();
+            }
```

Comment:
this is not obvious this will not loop.
Do 
```
Page page = rnn(pageSource.getNextPage())
```

Reply:
This loops to get the first non-null page.